linera-storage/src/db_storage.rs

```diff
@@ -499,13 +519,40 @@ where
         self.write_batch(batch).await
     }
 
-    async fn write_hashed_blob(
+    async fn write_hashed_blob(&self, blob: &HashedBlob) -> Result<(), ViewError> {
+        let mut batch = Batch::new();
+        self.add_blob_to_batch(&blob.id(), blob, &mut batch)?;
+        self.write_batch(batch).await?;
+        Ok(())
+    }
+
+    async fn maybe_write_blob_state(
         &self,
-        blob: &HashedBlob,
-        last_used_by: &CryptoHash,
+        blob_id: BlobId,
+        blob_state: BlobState,
+    ) -> Result<Epoch, ViewError> {
+        if self.contains_blob_state(blob_id).await? {
+            let current_blob_state = self.read_blob_state(blob_id).await?;
```

Comment:
Does that make two separate database queries? Would be more efficient to only try `read_blob_state`, and if it's a "not found" error we write it.

We could also deduplicate the two `write_blob_state` calls in this function.